### PR TITLE
Parse recently modified files even if just parsed

### PR DESCRIPTION
**Pull request by kaxil**
This commit adds an optimization where the recently modified files
(detected by mtime) will be parsed even though it has not reached
`min_file_process_interval`.

This way you can increase `[scheduler] min_file_process_interval` to
a higher value like `600` or so when you have large number of files to
avoid unnecessary reparsing if files haven't changed, while still making
sure that modified files are taken care of.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/utils/dag_processing.py [hunk 1]
```diff
@@ -1057,11 +1057,15 @@ def prepare_file_path_queue(self):
             else:
                 file_paths.append(file_path)
 
-            # Find file paths that were recently processed
+            # Find file paths that were recently processed to exclude them
+            # from being added to file_path_queue
+            # unless they were modified recently and parsing mode is "modified_time"
             last_finish_time = self.get_last_finish_time(file_path)
+            file_modified_time = timezone.make_aware(datetime.fromtimestamp(files_with_mtime[file_path]))
```
Comment (jedcunningham) [inline airflow/utils/dag_processing.py]: Won't this raise a `KeyError` if we aren't in mtime mode?

Comment (kaxil) [reply airflow/utils/dag_processing.py]: Updated the logic

File: airflow/utils/dag_processing.py [hunk 2]
```diff
@@ -1052,18 +1052,25 @@ def prepare_file_path_queue(self):
...
+            # unless they were modified recently and parsing mode is "modified_time"
+            last_finish_time = self.get_last_finish_time(file_path)
             if is_mtime_mode:
                 files_with_mtime[file_path] = os.path.getmtime(file_path)
+                file_modified_time = timezone.make_aware(datetime.fromtimestamp(files_with_mtime[file_path]))
+
+                # If file is modified recently we want to process it again
+                # ignoring "self._file_process_interval" (min_file_process_interval)
+                if last_finish_time is not None and (file_modified_time < last_finish_time):
+                    file_paths_recently_processed.append(file_path)
```
Comment (jedcunningham) [inline airflow/utils/dag_processing.py]: This doesn't honor `file_process_interval`, or am I missing something?

Comment (jedcunningham) [reply airflow/utils/dag_processing.py]: I'm thinking the code block below needs to come out a level so it runs regardless of mode.

**Review by jedcunningham — APPROVED** [review]

**Review by ephraimbuddy — APPROVED** [review]

**Pull request merged**